content/src/logic/delete-unreferenced-files.ts

```diff
@@ -11,46 +12,61 @@ export async function deleteUnreferencedFiles(
+      let totalAddedHashes = 0
+      for await (const hash of streamOfHashes) {
+        totalAddedHashes++
+        referencedHashesBloom.add(hash)
+      }
+      return totalAddedHashes
     }
-    return totalAddedHashes
-  }
 
-  await runLoggingPerformance(logger, 'populate bloom filter', async () => {
-    const totalEntityIds = await runLoggingPerformance(
-      logger,
-      'add stream of entity ids to bloom filter',
-      async () => await addAllToBloomFilter(streamAllDistinctEntityIds(components))
-    )
-
-    const totalContentFileHashes = await runLoggingPerformance(
-      logger,
-      'add of stream content file hashes to bloom filter',
-      async () => await addAllToBloomFilter(streamAllDistinctContentFileHashes(components))
-    )
-    logger.info(`Created bloom filter with ${totalEntityIds} entity ids and ${totalContentFileHashes} content hashes.`)
-  })
-
-  const queue = new PQueue({ concurrency: 1000 })
-  let numberOfDeletedFiles = 0
-  logger.info(`Deleting files...`)
-  for await (const storageFileId of components.storage.allFileIds()) {
-    if (!referencedHashesBloom.has(storageFileId)) {
-      await queue.add(async () => {
-        try {
-          await components.storage.delete([storageFileId])
-          numberOfDeletedFiles++
-        } catch (error) {
-          logger.error(error, { storageFileId })
-        }
-      })
+    await runLoggingPerformance(logger, 'populate bloom filter', async () => {
+      const totalEntityIds = await runLoggingPerformance(
+        logger,
+        'add stream of entity ids to bloom filter',
+        async () => await addAllToBloomFilter(streamAllDistinctEntityIds(components))
+      )
+
+      const totalContentFileHashes = await runLoggingPerformance(
+        logger,
+        'add of stream content file hashes to bloom filter',
+        async () => await addAllToBloomFilter(streamAllDistinctContentFileHashes(components))
+      )
+      logger.info(
+        `Created bloom filter with ${totalEntityIds} entity ids and ${totalContentFileHashes} content hashes.`
+      )
+    })
+
+    const queue = new PQueue({ concurrency: 1000 })
+    let numberOfDeletedFiles = 0
+
+    logger.info(`Deleting files...`)
+    for await (const storageFileId of storage.allFileIds()) {
+      if (!referencedHashesBloom.has(storageFileId)) {
+        await queue.add(async () => {
+          try {
+            await components.storage.delete([storageFileId])
+            numberOfDeletedFiles++
+          } catch (error) {
+            logger.error(error, { storageFileId })
+          }
+        })
+      }
     }
+    logger.info(`Deleted ${numberOfDeletedFiles} files`)
+  } else {
+    logger.error('deleteUnreferencedFiles only works with FolderBasedContentStorage')
   }
-  logger.info(`Deleted ${numberOfDeletedFiles} files`)
+}
+
+function isFolderBasedStorage(x: any): x is FolderBasedContentStorage {
```

Comment:
Could we move this responsibility to the `catalyst-storage` ?

Reply:
which responsibility? the helper function?

Reply:
exactly